protobuf/model_config.proto

```diff
@@ -509,6 +509,28 @@ message BatchInput
     //@@         The data of the tensor will be uninitialized.
     //@@
     BATCH_MAX_ELEMENT_COUNT_AS_SHAPE = 3;
+
+    //@@      .. cpp:enumerator:: Kind::BATCH_ITEM_SHAPE = 4
+    //@@
+    //@@         Among the requests in the batch, the shape of the
+    //@@         'source_input' will be added as input with shape
+    //@@         [batch_size, len(input_dim)]. For example, if one
+    //@@         batch-2 input with shape [3, 1] and batch-1 input
+    //@@         with shape [2, 2] are batched, the batch input will
+    //@@         have shape [3, 2] and value [ [3, 1], [3, 1], [2, 2]].
+    //@@
+    BATCH_ITEM_SHAPE = 4;
+
+    //@@      .. cpp:enumerator:: Kind::BATCH_ITEM_SHAPE_FLATTEN = 5
+    //@@
+    //@@         Among the requests in the batch, the shape of the
+    //@@         'source_input' will be added as input with shape
+    //@@         [batch_size * len(input_dim)]. For example, if one
```

Comment:
Fix to mention flat structure of shape

Reply:
Can you clarify this comment?

Reply:
I think we must explicitly call out single dimensionality of tensor. It is hard to ascertain that with just [batch_size * len(input_dim)]